src/App.tsx

```diff
@@ -26,70 +16,125 @@ type ReorderOptions = {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
-}
+};
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+ type State = {
+   isReversed: boolean,
+   sortType: SortType,
+ };
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  handleSortByAlpabet = () => {
+    this.setState({ sortType: SortType.ALPABET });
+  };
+
+  handleSortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  NONE = () => {
+    this.setState({ sortType: SortType.NONE });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+
+    const preparedGoods
+    = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classnames(
+              'button is-info', {
+                'is-light': sortType !== SortType.ALPABET,
+              },
+            )}
+            onClick={this.handleSortByAlpabet}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classnames(
+              'button is-success', {
+                'is-light': sortType !== SortType.LENGTH,
+              },
+            )}
+            onClick={this.handleSortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classnames(
+              'button is-warning', {
+                'is-light': sortType !== SortType.NONE,
+              },
+            )}
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+
+          {(sortType !== SortType.NONE || isReversed)
```

Comment:
```suggestion
          {(sortType !== SortType.NONE || isReversed)
```
pass `(sortType !== SortType.NONE || isReversed)` condition into separate variable